### DISCUSSION TITLE
Get target name

### DISCUSSION THREAD

**Discussion by caxapexac** [parcel-bundler/parcel#8639] [Q&A]
I'm looking for `process.env.TARGET` string to make some conditional things based on the target type but can't find it.
I've come up with having different html entry in every target and declare target variables here but its pretty ugly

Not related: https://github.com/parcel-bundler/parcel/discussions/8252 (its about plugin and Im asking about runtime)

**Comment by mischnic** [comment]
It's the situation as in https://github.com/parcel-bundler/parcel/discussions/8252#discussioncomment-3069599.
process.env replacement happens in a transformer, and there's no unique target in a transformer as there can be multiple.

**Reply by caxapexac** [reply]
But Im not asking for `process.env`, Im asking for any other way of getting this value, it could be macro with hardcode replacement, could be virtual library (imagine `import {targetName} from '$$currentTarget.js'`), could be import.meta.target, could be ftp::/localhost:9999/getTargetName (joking of course), could be literally anything. Is there any way of passing meaningful data into the build process?
